File: DirWalk.java
import java.io.*;

  
public class DirWalk {

	public static int dirwalk(String path)
	{
		String[] dirlist = (new File(path)).list();
		int total = 0;

		if (dirlist == null) {
			return 0;
		} 
		for (int i=0; i < dirlist.length; i++) {
			String filename = dirlist[i];
			if (filename.equals(".") || filename.equals("..")) {
				continue;
			}
			String fullpath=path + "/" + filename;
			File newfile = new File(fullpath);
			if (newfile.isDirectory()) {
				total += dirwalk(fullpath);
			}
			else {
				total++;
			}
		}
		return total;
	}


	public static void main(String args[])
	{
		int total;
		String dirname = args[0];
  
		System.out.println("Java Traversing: " + args[0]);
		total = dirwalk(args[0]);
		System.out.println("Java Total Files: " +  total);
	}
  
}

File: dirwalk.rb
#!/usr/bin/env ruby
#
def dirwalk(path)
	total = 0

	Dir.entries(path).each do |entry|
		next if entry == "." || entry == ".."
		fullpath = path + "/" +  entry
		if File.directory?(fullpath)
			total = total + dirwalk(fullpath)
		else
			total = total + 1
		end
	end	
	return total
end

if (ARGV.count == 0)
   puts "Usage: dirwalk.rb DIRECTORY"
   exit
end


puts "Ruby Traversing: #{ARGV[0]}"
total = dirwalk(ARGV[0])
puts "Ruby Total Files: #{total}"
